cmd/minikube/cmd/start.go

```diff
@@ -477,16 +489,57 @@ func showVersionInfo(k8sVersion string, cr cruntime.Manager) {
 	}
 }
 
+/**
+Function to check for kubectl. The checking is to compare
+the version reported by both the client and server. It is
+that kubectl will be of the same version or 1 lower than
+the kubernetes version.
+*/
 func showKubectlConnectInfo(kcs *kubeconfig.Settings) {
+	var output []byte
+	clientVersion := kubectlversion{}
+	serverVersion := kubectlversion{}
+
+	path, err := exec.LookPath("kubectl")
+
+	if err != nil {
+		out.T(out.Tip, "For best results, install kubectl: https://kubernetes.io/docs/tasks/tools/install-kubectl/")
+	} else {
+		glog.Infof("Kubectl found at the following location %s. Let's execute and get the result", path)
+		output, err = exec.Command(path, "version", "--output=json").Output()
+
+		// ...once we get something back do some version checking
+		if err == nil {
+			glog.Infof("Received output from kubectl %s", output)
+
+			// unmarshal both the {client}
+			clientjsonErr := json.Unmarshal(output, &clientVersion)
+
+			// .......  and {server} json
+			serverjsonErr := json.Unmarshal(output, &serverVersion)
+
+			if (clientjsonErr != nil) || (serverjsonErr != nil) {
+				glog.Infof("There was an error processing the json output")
+			} else {
+				// obtain the minor version for both
+				serverMinor, _ := strconv.Atoi(serverVersion.SVersion.Minor)
+				clientMinor, _ := strconv.Atoi(serverVersion.CVersion.Minor)
+
+				if clientMinor < serverMinor {
```

Comment:
This is not quite correct yet. https://kubernetes.io/docs/setup/release/version-skew-policy/#kubectl dictates that you are allowed up to one version difference.  One possibility is:

`if math.Abs(float64(clientMinor - serverMinor)) > 1 {`